File: C++/SelectionSort.cpp

#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;


void selectionSort(int arr[], int size) {
	
	for (int i = 0; i < size-1; i++) {
		int min = i;
		for (int j = i+1; j < size; j++) {
			if (arr[j] < arr[min]) 
				min = j;
		}
		int temp = arr[i];
		arr[i] = arr[min];
		arr[min] = temp;
	}
}



int main() {
	
	const int MAX_ARRAY_SIZE = 10;
	int arr[MAX_ARRAY_SIZE];
	srand(time(NULL));
	
	for (int i = 0; i < MAX_ARRAY_SIZE; i++) {
		arr[i] = rand() % 100;
	}
	
	cout << "ARRAY BEFORE SELECTION SORT APPLIED" << endl;
	for (int i = 0; i < MAX_ARRAY_SIZE; i++) {
		cout << arr[i] << ' ';
	}
	
	cout << endl;
	
	selectionSort(arr, MAX_ARRAY_SIZE);
	
	cout << "ARRAY AFTER SELECTION SORT APPLIED" << endl;
	for (int i = 0; i < MAX_ARRAY_SIZE; i++) {
		cout << arr[i] << ' ';
	}
	
	cout << endl;
}

File: JavaScript/Selection-Sort.js
const MAX_ARR_SIZE = 10;

main();

function main(){
    let arr = []

    for (i = 0; i < MAX_ARR_SIZE; i++)
        arr[i] = Math.floor(Math.random() * 101)
    
    console.log(arr)

    selectionSort(arr)

    console.log(arr)
}

function selectionSort(arr){
    let n = arr.length
    if(n == 0) return;

    for (i = 0; i < n-1; i++){
        min_index = i
        for(j = i+1; j < n; j++){
            if(arr[j] < arr[min_index])
                min_index = j
        }
        let temp = arr[i]
        arr[i] = arr[min_index]
        arr[min_index] = temp
    }
}